### PR TITLE
Data in models

### PR DESCRIPTION
**Pull request by getssh**
Hello, I have worked on processing data in the models task

- Created necessary models for all tables
- Used rails c to interact with the database
- Created users, posts, comments using necessary command in rails c
- Created all required custom method for all model, classes
- Managed all associations between tables (one to one, many to many)
- Checked for linter errors

sample_rails_console_output

Thank you

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+class User < ApplicationRecord
+  has_many :posts, foreign_key: :author_id
+
```
Comment (TadesseDev) [inline app/models/user.rb]: - [x] Dear @getssh, As dispict on the [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png) and as stated in the project requirement, the suer `has many Likes and comments` beside posts. Please make sure to add the `has_many :Like` and `has_many :comment` inside the user model.

File: app/models/user.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  def recent_posts
+    posts.order(created_at: :desc).limit(3)
+  end
+
+  def update_posts_counter
+    update(post_counter: posts.count)
+  end
```
Comment (TadesseDev) [inline app/models/user.rb]: - [x] Dear @getssh, Please notice this (`update_posts_counter`) method is expected to be implemented inside the `Post` model.

Comment (TadesseDev) [inline app/models/user.rb]: - [Optional] To make your model methods even more professional you can use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_post_counter` method every time a new post has been saved to the database, then make the `update_post_counter` method private. This same approach can be used for calling `update_comments_counter` and `update_likes_counter` methods.

**Review by TadesseDev — CHANGES_REQUESTED** [review]
### Hi @getssh 

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
212094426-b41dd457-c9db-4ab3-bbe9-176dd8dd08b7




## Highlights
 - [x]  Correct GitFlow  ✅
 - [x]  Good commit history  ✅
 - [x]  Good PR title/description  ✅
 - [x]  All linters passed  ✅
 
## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by getssh** [comment]
@TadesseDev , Thank you for the review. I have Implemented the requested changes.

**Review by franclobo — APPROVED** [review]
Hi @getssh, 

# STATUS: APPROVED 😃 

git-merge

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- No linter errors ✔️ 
- Descriptive PR ✔️ 
- Correct Git flow ✔️ 
- Correct asossiations ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by getssh** [comment]
Thank you for the review

**Pull request merged**